### PR TITLE
Prevent replacing constant extents with non-const in concretization

### LINKED ISSUE #1572: `lessThan` ran into infinite recursion.
```
$ git fetch origin wjy/recurse
$ git checkout wjy/recurse
$ _bn && NVFUSER_DISABLE=parallel_compile python repro.py
```

`NVFUSER_DISABLE=parallel_compile` is there to display the call stack better. 

To make debugging easier, I limited recursion of `lessEqual` to 100 levels. The reproducer is still valid but slower with that check removed.

### PR DESCRIPTION
**Pull request by jacobhinkle**
Fixes #1572

In that case, `concretizeReshape` is replacing some extents in the new shape, such as `(nvfuser_index_t)16` with non-constant values like `i0`. This leads to a downstream expression that originally was constant `0` to be non-constant, so it cannot be picked up by `RemoveEmptyPass`. This PR avoids concretizing constant extents with non-constants. This should be safe since even if those tensors are segmentation edges, we will not lose any scalars because the extent is constant so there are no free scalars to lose.

### CODE REVIEW

**Review by wujingyue — APPROVED** [review]
LGTM - pending a unit test. Thanks for the quick fix, @jacobhinkle

File: test/test_dynamic_transform.cpp [hunk 1]
```diff
@@ -1295,4 +1295,51 @@ TEST_F(NVFuserTest, SymbolicSqueeze) {
...
+          castOp(DataType::Index, IrBuilder::create<Val>(4096, DataType::Int)),
+          castOp(DataType::Index, IrBuilder::create<Val>(128, DataType::Int)),
+      });
+  // Note this slice has zero extent in last dimension. RemoveEmptyPass should
+  // recognize this and replace with full()
+  auto tv5 = slice(tv4, {0, 0, 0}, {32, 4096, 0});
+
+  fusion->addOutput(tv5);
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
```
Comment (jacobhinkle) [inline test/test_dynamic_transform.cpp:1334]: Without the change to `dynamic_transform.cpp` this hangs with infinite recursion due to contradiction `0 < blockDim.x < 0`.

**Comment by jacobhinkle** [comment]
!build --diff

**Comment by jacobhinkle** [comment]
> LGTM - pending a unit test. Thanks for the quick fix, @jacobhinkle

Happy to help! Thanks for finding this subtle bug.

**Comment by xwang233** [comment]
!build --diff

**Comment by jacobhinkle** [comment]
!build --diff

**Pull request merged**